src/ex1/PokemonClient.js

```diff
@@ -0,0 +1,44 @@
+class PokemonClient {
+    constructor() {
+        this.API_BASE = "https://pokeapi.co/api/v2";
+    }
+
+    async getPokemon(pokemonId) {
+        try {
+            const response = await fetch(`${this.API_BASE}/pokemon/${pokemonId}`);
+            const result = await response.json();
+            return result.name;
+        } catch (error) {
+            const errorMessage = `Pokemon with ID ${pokemonId} was not found`;
+            return errorMessage;
+        }
+    }
+
+    async getAllPokemons(pokemonsIds) {
+        const allPromises = [];
+        let pokemons = [];
+        try {
+            pokemonsIds.forEach((pokemonId) => {
+                allPromises.push(fetch(`${this.API_BASE}/pokemon/${pokemonId}`));
+            });
+
+            await Promise.all(allPromises).then((responses) => {
+                responses.some((response) => {
+                    if (!response.ok) {
+                        const errorMessage = `Failed to fetch pokemon with this input ${pokemonsIds}`;
+                        pokemons = errorMessage;
+                        return pokemons;
+                    }
+                    const result = response.json();
+                    result.then((pokemon) => {
+                        pokemons.push(pokemon.name);
+                    });
+                });
+            });
+            return pokemons;
+        } catch (error) {
+            const errorMessage = `Failed to fetch pokemon with this input ${pokemonsIds}`;
```

Comment:
Extra thought: you have 2 methods - get a single pokemon and get all pokemons
if you think about it - get all pokemon is just get a single pokemon over and over again
you should reuse the `await fetch` part - it should appear only once which makes error handling easier as well
Reusing code is best practice - if you want help with this please send a message